mypy/subtypes.py

```diff
@@ -333,7 +333,7 @@ def unify_generic_callable(type: CallableType, target: CallableType,
         return None
     msg = messages.temp_message_builder()
     applied = mypy.applytype.apply_generic_arguments(type, inferred_vars, msg, context=target)
-    if msg.is_errors() or not isinstance(applied, CallableType):
+    if applied is None or msg.is_errors():
```

Comment:
`not isinstance(applied, CallableType)` before if and only if `applied is None` now.
`not msg.iserrors()` also implies `not isinstance(applied, AnyType) ` and therefore `isinstance(applied, CallableType)`.
Am I wrong?